### PR TITLE
Temp changes

### PR DESCRIPTION
**Pull request by bajormar**
Created new component

### CODE REVIEW

**Review by Gedzis — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -1,7 +1,26 @@
...
+import List from './List';
+import { useState } from 'react';
 
 function App() {
-  return <div>App</div>;
+  const [page, change] = useState(0);
+  const [data] = useState([
+    { name: 'item1', hidden: true },
+    { name: 'item2', hidden: false },
+  ]);
```
Comment (tadasgo) [inline src/App.tsx:10]: no setData, so either you don't need state or you are doing something wrong 😄

File: src/List.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+
+const List = ({ data, setData }: { data: any[]; setData: () => void; page: number }) => {
+  const [items, setItems] = useState<{ name: string }[]>([]);
```
Comment (tadasgo) [inline src/List.tsx:4]: unnecessary state

**Review by benz0is — APPROVED** [review]

File: src/List.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+
+const List = ({ data, setData }: { data: any[]; setData: () => void; page: number }) => {
```
Comment (tadasgo) [inline src/List.tsx:3]: accepts page but doesn't use it. `Any[]` is too abstract of a type

File: src/List.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  }, []);
+
+  return (
+    <div>
+      {items.map((a) => (
+        <div>{a.name}</div>
+      ))}
+      <button
+        onClick={() => {
+          setData();
```
Comment (tadasgo) [inline src/List.tsx:17]: onClick={setData}

File: src/App.tsx [hunk 5]
```diff
@@ -1,7 +1,26 @@
 import './App.css';
+import List from './List';
+import { useState } from 'react';
 
 function App() {
-  return <div>App</div>;
+  const [page, change] = useState(0);
```
Comment (tadasgo) [inline src/App.tsx:6]: weird naming `[page, setPage]`

File: src/List.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+  useEffect(() => {
+    setItems(data.filter((i) => !i.isHidden));
+  }, []);
+
+  return (
+    <div>
+      {items.map((a) => (
+        <div>{a.name}</div>
+      ))}
```
Comment (tadasgo) [inline src/List.tsx:14]: You can check for nonHidden items inside the map fn and avoid another state inside list component

File: src/App.tsx [hunk 7]
```diff
@@ -1,7 +1,26 @@
...
+  ]);
+
+  return (
+    <div>
+      <List
+        page={page}
+        data={data}
+        setData={() => {
+          data.push({ name: 'item3', hidden: true });
+          change((p) => p + 1);
```
Comment (tadasgo) [inline src/App.tsx:19]: Also this is kind of messed up. You give component a setData fn which secretly changes the page behind the scenes. I would separate these two

**Pull request closed without merging**